### ISSUE TITLE
Fabric Node open failed with error code = E_ACCESSDENIED

### ISSUE THREAD

**Issue by knizkar** [Azure/service-fabric-issues#1056]
After installation of Microsoft Azure Service Fabric SDK version 3.1.274 on Windows 10 with latest updates local dev cluster fails to start. Problem seems to be in `Fabric.exe` process which immediately exits.

Event log:

```xml
<Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">
<System>
    <Provider Name="Microsoft-ServiceFabric" Guid="{CBD93BC2-71E5-4566-B3A7-595D8EECA6E8}"/>
    <EventID>256</EventID>
    <Version>1</Version>
    <Level>2</Level>
    <Task>1</Task>
    <Opcode>0</Opcode>
    <Keywords>0x8000000000000001</Keywords>
    <TimeCreated SystemTime="2018-05-14T18:40:15.935568600Z"/>
    <EventRecordID>604468</EventRecordID>
    <Correlation/>
    <Execution ProcessID="3388" ThreadID="17536"/>
    <Channel>Microsoft-ServiceFabric/Admin</Channel>
</System>
<EventData>
    <Data Name="id"/>
    <Data Name="type">CryptoUtility</Data>
    <Data Name="text">CertCreateSelfSignCertificate failed: E_ACCESSDENIED</Data>
</EventData>
</Event>
<Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">
<System>
    <Provider Name="Microsoft-ServiceFabric" Guid="{CBD93BC2-71E5-4566-B3A7-595D8EECA6E8}"/>
    <EventID>25600</EventID>
    <Version>1</Version>
    <Level>2</Level>
    <Task>100</Task>
    <Opcode>0</Opcode>
    <Keywords>0x8000000000000001</Keywords>
    <TimeCreated SystemTime="2018-05-14T18:40:15.935572800Z"/>
    <EventRecordID>604469</EventRecordID>
    <Correlation/>
    <Execution ProcessID="3388" ThreadID="17536"/>
    <Channel>Microsoft-ServiceFabric/Admin</Channel>
</System>
<EventData>
    <Data Name="id">a3784be1d81710242ed0a9632647b4f7</Data>
    <Data Name="type">Security</Data>
    <Data Name="text">ipcServer-&gt;SecuritySettings.CreateSelfGeneratedCertSslServer error=S_OK</Data>
</EventData>
</Event>
<Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">
<System>
    <Provider Name="Microsoft-ServiceFabric" Guid="{CBD93BC2-71E5-4566-B3A7-595D8EECA6E8}"/>
    <EventID>2560</EventID>
    <Version>1</Version>
    <Level>2</Level>
    <Task>10</Task>
    <Opcode>0</Opcode>
    <Keywords>0x8000000000000001</Keywords>
    <TimeCreated SystemTime="2018-05-14T18:40:15.935863100Z"/>
    <EventRecordID>604470</EventRecordID>
    <Correlation/>
    <Execution ProcessID="3388" ThreadID="17536"/>
    <Channel>Microsoft-ServiceFabric/Admin</Channel>
</System>
<EventData>
    <Data Name="id"/>
    <Data Name="type">FabricExe</Data>
    <Data Name="text">Fabric Node open failed with error code = E_ACCESSDENIED</Data>
</EventData>
</Event>
```

**Comment by knizkar** [comment]
Problem can be worked around by setting Hosting/FabricContainerAppsEnabled to `false` in cluster manifest.

*discovered by consulting Service Fabric [source code ](https://github.com/Microsoft/service-fabric/blob/master/src/prod/src/FabricNode/FabricNode.cpp#L1463-L1487)

**Comment by vturecek** [comment]
@knizkar just as a sanity check, do you have:
 - containers enabled in Windows?
 - Docker for Windows installed? If so,
     - switched to Windows containers?
     - what version?

**Comment by aloneguid** [comment]
I have the same problem. This particular machine has containers feature enabled, however neither docker not Hyper-V is enabled.

**Comment by knizkar** [comment]
Not sure if the problem is related to wrong detection of container support, adding more details from Process Monitor filtering on all `ACCESS_DENIED` events:

![image](https://user-images.githubusercontent.com/1428928/40668429-c7d90b06-6364-11e8-93c1-c5a44a76a730.png)

Stacktrace:
![image](https://user-images.githubusercontent.com/1428928/40668470-e378dbca-6364-11e8-9bce-3f5948004638.png)

**Comment by DrunkenHusky** [comment]
Hi Guys

This also happens to me toaday. The fix works also, if you try to install a Cluster via Powershell.
My Enviroment is a Windows Server 2016 Standard with the 2018-05 Cumulative Updates and some other installed. No Problem on a Server 2016 without these updates (before 2017-11). If Service Fabric is installed on the machine it looks like I'm not able to install the Windows Updates.

Example JSON:
`{
    "name": "SampleCluster",
    "clusterConfigurationVersion": "1.0.0",
    "apiVersion": "10-2017",
    "nodes": [
      {
        "nodeName": "vm0",
        "iPAddress": "localhost",
        "nodeTypeRef": "NodeType0",
        "faultDomain": "fd:/dc1/r0",
        "upgradeDomain": "UD0"
      },
      {
        "nodeName": "vm1",
        "iPAddress": "localhost",
        "nodeTypeRef": "NodeType1",
        "faultDomain": "fd:/dc2/r0",
        "upgradeDomain": "UD1"
      },
      {
        "nodeName": "vm2",
        "iPAddress": "localhost",
        "nodeTypeRef": "NodeType2",
        "faultDomain": "fd:/dc3/r0",
        "upgradeDomain": "UD2"
      }
    ],

    "properties": {
        "diagnosticsStore": 
        {
            "metadata":  "Please replace the diagnostics file share with an actual file share accessible from all cluster machines. For example, \\\\localhost\\DiagnosticsStore.",
            "dataDeletionAgeInDays": "21",
            "storeType": "FileShare",
            "connectionstring": "c:\\ProgramData\\SF\\DiagnosticsStore"
        },
        "nodeTypes": [
          {
            "name": "NodeType0",
            "clientConnectionEndpointPort": "19000",
            "clusterConnectionEndpointPort": "19001",
            "leaseDriverEndpointPort": "19002",
            "serviceConnectionEndpointPort": "19003",
            "httpGatewayEndpointPort": "19080",
            "reverseProxyEndpointPort": "19081",
            "applicationPorts": {
                "startPort": "20001",
                "endPort": "20031"
            },
            "isPrimary": true
          },
          {
            "name": "NodeType1",
            "clientConnectionEndpointPort": "19004",
            "clusterConnectionEndpointPort": "19005",
            "leaseDriverEndpointPort": "19006",
            "serviceConnectionEndpointPort": "19007",
            "httpGatewayEndpointPort": "19082",
            "reverseProxyEndpointPort": "19083",
            "applicationPorts": {
                "startPort": "20288",
                "endPort": "20318"
            },
            "isPrimary": true
          },
          {
            "name": "NodeType2",
            "clientConnectionEndpointPort": "19008",
            "clusterConnectionEndpointPort": "19009",
            "leaseDriverEndpointPort": "19010",
            "serviceConnectionEndpointPort": "19011",
            "httpGatewayEndpointPort": "19084",
            "reverseProxyEndpointPort": "19085",
            "applicationPorts": {
                "startPort": "20575",
                "endPort": "20605"
            },
            "isPrimary": false
        }
      ],
      "fabricSettings": [
        {
          "name": "Setup",
          "parameters": [
            {
                "name": "FabricDataRoot",
                "value": "C:\\ProgramData\\SF"
            },
            {
                "name": "FabricLogRoot",
                "value": "C:\\ProgramData\\SF\\Log"
            }
          ]
        },
        {
        "name": "Hosting",
                "parameters": [
                    {
                        "name": "FabricContainerAppsEnabled",
                        "value": "false"
                    }
                ]
            }
      ]
    }
}

`

Hope this helps someone else. I just lost 3 days of work because of that bug. :-(

Cheers

**Comment by dragav** [comment]
For local/devbox clusters which do not specify any certificates (unsecured), Fabric will attempt to create a self-signed certificate with a key stored in the machine key store. The code runs with lowered privileges, and so receives the access denied error mentioned above. 
The workarounds would be to run the cluster elevated, or secure the cluster with certificates accessible to the user account it's running as. We're investigating a fix (to fall back to the user key store for self-signed certs.)

**Comment by jitendrasankhla** [comment]
I am having the same issue, instead when I tried the workaround as suggested in #1056, it didn't work. 
In addition, I am getting `client-MY-MACHINE:19000/***.***.***.**:19000: error = 2147943625, failureCount=975. Filter by (type~Transport.St && ~"(?i)MY-MACHINE:19000") to get listener lifecycle. Connect failure is expected if listener was never started, or listener/its process was stopped before/during connecting.` and `client-MY-MACHINE:19000 : connect failed, having tried all addresses` from Service-Fabric windows event logs. Also when I tried to update _C:\Program Files\Microsoft SDKs\Service Fabric\ClusterSetup\NonSecure\OneNode\ClusterManifestTemplate.json_  I didn't find any 
`         {
            "name": "FabricContainerAppsEnabled",
            "value": "false"
          }` 
in the file so that it could be set to false. 

I have tried re-installing Service Fabric SDK, my current version of SDK is 3.2.162.9494

**Comment by jitendrasankhla** [comment]
The only thing that helped - going back to 6.2.283/3.1.283

**Comment by dragav** [comment]
@jitendrasankhla the issue you're describing seems to be a different one - the error code in your case is ERROR_CONNECTION_REFUSED, which indicates a server-side problem. Please open a separate issue to track this, and provide logs if you can. Thanks.

**Comment by masnider** [comment]
To folks encountering this issue: if disabling the containers features is not an option, there should be an additional workaround. 

First: Let's confirm it is a permissions issue by running the following from an elevated PowerShell prompt:

```(Get-Acl -Path ($env:ProgramData+'\Microsoft\Crypto\rsa\MachineKeys\')).Access | select FileSystemRights,AccessControlType,IdentityReference | ft -AutoSize```
 
You should see something like this:
```
        FileSystemRights     AccessControlType     IdentityReference
        ----------------     -----------------     -----------------
   Write, Read, Synchronize             Allow       Everyone
ReadAndExecute, Synchronize             Allow       NT AUTHORITY\NETWORK SERVICE
                FullControl             Allow       BUILTIN\Administrators
```

We suspect ‘everyone’ is not on your list, or is missing the ‘write’ permission. The workaround (besides suppressing FabricContainerApps) is to explicitly add that permission:
 
```
$sid="*S-1-1-0" #everyone
$path=$env:ProgramData+'\Microsoft\Crypto\Rsa\MachineKeys' #machine key store
$perms="(RX,W)" # read, write and execute
icacls $path /grant $sid`:$perms
```
 
(note the backtick “`” character above, it's necessary.)
@dragav fyi in case I screwed up transcribing this :)

**Comment by jxnuzhangwen** [comment]
this issue torment me two day! thanks god! it work for me below! 

{
            "name": "FabricContainerAppsEnabled",
            "value": "false"
}

**Comment by ikaros88** [comment]
@masnider 
I have tried everything on one of our dev pcs that can not seem to be able to setup a local cluster.

I ran your ps script and have the following output after adding the Write permission for everyone:
```
                  FileSystemRights AccessControlType IdentityReference
                  ---------------- ----------------- -----------------
Write, ReadAndExecute, Synchronize             Allow Everyone
                       FullControl             Allow NT AUTHORITY\SYSTEM
                       FullControl             Allow BUILTIN\Administrators
       ReadAndExecute, Synchronize             Allow BUILTIN\Users
       ReadAndExecute, Synchronize             Allow Everyone
```
However, even after restarting, i am still receiving

CertCreateSelfSignCertificate failed: E_ACCESSDENIED
ipcServer->SecuritySettings.CreateSelfGeneratedCertSslServer error=S_OK
Fabric Node open failed with error code = E_ACCESSDENIED

Deactivating the FabricContainerAppsEnabled setting has no effect either.

Are there any further suggestions? I have reinstalled SF multiple times by now. Most of our other dev machines work fine.

**Comment by masnider** [comment]
@ikaros88 - can you share the full setup logs from this environment? We haven't see a case where these workarounds were not successful, so I'm wondering if you're running into something else. 

(assuming Windows) The reg key HKLM\SOFTWARE\Microsoft\ServiceFabric\FabricLogRoot should point to the location of the logs. Zip the directory and attach the file here; you can also zip and email it to me (masnider, or @dragav, dragosav at microsoft if you want)

**Comment by rnilsark** [comment]
We got this problem on almost all dev computers when upgrading to SF 3.2.176.
Adding write permission for NETWORK SERVICE or even removing the security config for NETWORK SERVICE resolves the issue ("Everyone" has Write permission). 

One dev computer was set up from scratch with SF 3.2.176 as first the installed SDK. Not sure, but on this computer I think this issue emerged when using the Refresh debug application mode (which creates a couple security groups). I'm pretty sure it worked starting a cluster before that, so maybe a combination the activating Refresh and the update.

By the way, running a secure cluster.

**Comment by chandimar001** [comment]
I started running in to this yesterday but haven't found a resolution.
@rnilsark How did you add write permission for NETWORK SERVICE? Was it through the Powershell script @masnider provided above? Here is what I currently have:
>            FileSystemRights AccessControlType IdentityReference
>           ---------------- ----------------- -----------------
>   Write, Read, Synchronize             Allow Everyone
>                FullControl             Allow BUILTIN\Administrators
>ReadAndExecute, Synchronize             Allow MYDOMAIN\ztest01

I'm hoping that getting NETWORK SERVICE on that list fixes the issue.

**Comment by chandimar001** [comment]
I have attached Logs and Fabric Setup trace in case it helps.

[FabricSetup-131800552523653998.zip](https://github.com/Azure/service-fabric-issues/files/2334127/FabricSetup-131800552523653998.zip)
[Log.zip](https://github.com/Azure/service-fabric-issues/files/2334126/Log.zip)

**Comment by chandimar001** [comment]
My issue is resolved! Thanks for the pointers @rnilsark! 

I gave NETWORK SERVICE (SID S-1-5-20) access through the Powershell script. It added the group but cluster creation still failed. I then manually gave "Everyone" and "NETWORK SERVICE" Full Control to C:\ProgramData\Microsoft\Crypto\RSA\MachineKeys, rebooted my machine and created the cluster using DevClusterSetup.ps1 and it worked this time.

I am unblocked for development because VS 2017 can successfully deploy my code for debugging purposes. 

The only weird remaining issue is that I get a ERR_SSL_PROTOCOL_ERROR when I go to https://localhost:19080/Explorer/.

**Comment by ikaros88** [comment]
@chandimar001 thanks! you solved my problem as well. adding full control for everyone and network service solved this for me.

**Comment by aloneguid** [comment]
Giving full permissions to network service also solves the issue for me. It's not ideal though from security perspective.

**Comment by marclallen** [comment]
I'm having the same "No cluster endpoint is reachable,...".  I've done everything EXCEPT enable access for Everyone to the machine keys.  I really don't want to do that.  There's got to be a better way, right?

**Comment by marclallen** [comment]
I tried it.  Here's my output of the acl script:

```
(Get-Acl -Path ($env:ProgramData+'\Microsoft\Crypto\rsa\
MachineKeys\')).Access | select FileSystemRights,AccessControlType,IdentityReference | ft -AutoSize

FileSystemRights AccessControlType IdentityReference
---------------- ----------------- -----------------
     FullControl             Allow Everyone
     FullControl             Allow NT AUTHORITY\SYSTEM
     FullControl             Allow NT AUTHORITY\NETWORK SERVICE
       268435456             Allow BUILTIN\Administrators
     FullControl             Allow BUILTIN\Administrators
```

I still receive the following:

```
Local Cluster ready status: 92% completed.
Local Cluster ready status: 96% completed.
Local Cluster ready status: 100% completed.
WARNING: Service Fabric Cluster is taking longer than expected to connect.

Waiting for Naming Service to be ready. This may take a few minutes...
No cluster endpoint is reachable, please check if there is connectivity/firewall/DNS issue.
Connect-ServiceFabricCluster : No cluster endpoint is reachable, please check if there is connectivity/firewall/DNS
issue.
At C:\Program Files\Microsoft SDKs\Service Fabric\Tools\Scripts\ClusterSetupUtilities.psm1:644 char:12
+     [void](Connect-ServiceFabricCluster @connParams)
+            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [Connect-ServiceFabricCluster], FabricException
    + FullyQualifiedErrorId : TestClusterConnectionErrorId,Microsoft.ServiceFabric.Powershell.ConnectCluster
```
What can I provide you to help solve this?

**Comment by dharmeshtailor** [comment]
I copied over following folders from my colleague, (who still has old version fortunately) and replace it on my machine.

```
C:\Program Files\Microsoft Service Fabric
C:\Program Files\Microsoft SDKs\Service Fabric
```

Plus, I also modified the version in following registry keys

```
HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Service Fabric
HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Service Fabric SDK
```

It is working now properly and am not updating Service fabric till this issue is resolved.

Thanks.

**Comment by marclallen** [comment]
Wow.  Ok.. I'll try anything.  Any chance of you zipping those bad boys up and putting them on a drive share, somewhere?

**Comment by dharmeshtailor** [comment]
Unfortunately it worked for some time, but same issue again. This time I am getting following error in Event Log

```
Failed with Exception=System.Collections.Generic.KeyNotFoundException: The given key was not present in the dictionary.
   at System.ThrowHelper.ThrowKeyNotFoundException()
   at System.Collections.Generic.Dictionary`2.get_Item(TKey key)
   at System.Fabric.HostingSettings.LoadHostingSettings()
   at Hosting.ContainerActivatorService.HostingConfig..ctor()
   at Hosting.ContainerActivatorService.Program.Main(String[] args)
```

On further digging, I found this code System.fabric.HostingSettings.cs

```
IDictionary<string, string> dictionary = NativeTypes.FromNativeStringPairList(((NativeTypes.FABRIC_HOSTING_SETTINGS*) (void*) NativeContainerActivatorService.LoadHostingSettings().get_Result())->SettingsMap);
      HostingSettings hostingSettings = new HostingSettings();
      return new HostingSettings()
      {
        ContainerPackageRootFolder = dictionary["ContainerPackageRootFolder"],
        ContainerFabricBinRootFolder = dictionary["ContainerFabricBinRootFolder"],
        ContainerFabricLogRootFolder = dictionary["ContainerFabricLogRootFolder"],
        ContainerAppDeploymentRootFolder = dictionary["ContainerAppDeploymentRootFolder"],
        DockerRequestTimeout = TimeSpan.FromSeconds((double) int.Parse(dictionary["DockerRequestTimeout"])),
        ContainerImageDownloadTimeout = TimeSpan.FromSeconds((double) int.Parse(dictionary["ContainerImageDownloadTimeout"])),
        ContainerDeactivationTimeout = TimeSpan.FromSeconds((double) int.Parse(dictionary["ContainerDeactivationTimeout"])),
        ContainerDeactivationRetryDelayInSec = int.Parse(dictionary["ContainerDeactivationRetryDelayInSec"]),
        ContainerTerminationMaxRetryCount = int.Parse(dictionary["ContainerTerminationMaxRetryCount"]),
        ContainerEventManagerMaxContinuousFailure = int.Parse(dictionary["ContainerEventManagerMaxContinuousFailure"]),
        ContainerEventManagerFailureBackoffInSec = int.Parse(dictionary["ContainerEventManagerFailureBackoffInSec"]),
        EnableDockerHealthCheckIntegration = bool.Parse(dictionary["EnableDockerHealthCheckIntegration"]),
        ContainerServiceNamedPipeOrUnixSocketAddress = dictionary["ContainerServiceNamedPipeOrUnixSocketAddress"],
        DeadContainerCleanupUntilInMinutes = int.Parse(dictionary["DeadContainerCleanupUntilInMinutes"]),
        ContainerCleanupScanIntervalInMinutes = int.Parse(dictionary["ContainerCleanupScanIntervalInMinutes"])
      };
```

Any idea, where it is loading host settings from?

**Comment by dharmeshtailor** [comment]
Can we at least get hold of previous working versions? Microsoft Web Platform Installer does not show previous versions. Haven't done a single productive task since last 4 days. Just trying to resolve this issue.

**Comment by dragav** [comment]
@marclallen @dharmeshtailor I am truly sorry for the pain you're experiencing. If possible, please send me the SF traces, whose path you can find under $HKLM\Software\Microsoft\ServiceFabric\FabricLogRoot (I would only need the 'Traces' directory).

Please zip the directory and upload it here, or rename the zip and email me (dragosav at microsoft).

Alternatively, we can set up a phone call/online meeting and do a live investigation. I am in the Pacific Daylight Time (UTC-7), and can be available anytime from 8am to midnight or so.

**Comment by dharmeshtailor** [comment]
I have copied the folders from my colleague's machine. For me, it works for 5 minutes and then die. So every 5 minutes I run following command to make it work again:

.\CleanCluster.ps1
.\DevClusterSetup.ps1 -CreateOneNodeCluster -Auto

It fails with following error

```
Failed with Exception=System.Collections.Generic.KeyNotFoundException: The given key was not present in the dictionary.
   at System.ThrowHelper.ThrowKeyNotFoundException()
   at System.Collections.Generic.Dictionary`2.get_Item(TKey key)
   at System.Fabric.HostingSettings.LoadHostingSettings()
   at Hosting.ContainerActivatorService.HostingConfig..ctor()
   at Hosting.ContainerActivatorService.Program.Main(String[] args)

Warning : HostedServiceInstance:FabricContainerActivatorService terminated with exitcode 3762504530
```

[Traces.zip](https://github.com/Azure/service-fabric-issues/files/2471288/Traces.zip)

**Comment by dragav** [comment]
@dharmeshtailor in your case, the issue seems related to a mismatching of code vs config. Tracing back this thread, you mentioned that you copied over the SDK and Fabric folders from an older installation over a newer one. I don't think the cleanup was effective, and that would explain what you're seeing. 

Are you using containers? If not, disabling them may avoid this crash (which eventually terminates the cluster). Another question is whether you would consider going back to the most recent SF drop, and we'll work to get you past the certificate issue.

**Comment by dharmeshtailor** [comment]
Thank you for your response. How do I disable the container?

I'll update the SF to latest version on Monday and share the traces.

**Comment by dragav** [comment]
@dharmeshtailor Please see the comment from @knizkar earlier in this thread: "Hosting/FabricContainerAppsEnabled* to false in cluster manifest" 
{
  "name": "Hosting",
  "parameter": [
...
{
            "name": "FabricContainerAppsEnabled",
            "value": "false"
}]

**Comment by dharmeshtailor** [comment]
@dragav I already have that setting in config. I'll clean up and reinstall the latest version and then we try to resolve it from there.

Thanks.

**Comment by dharmeshtailor** [comment]
@dragav I have updated the service fabric to latest version and attached are the traces. I saw one new error:

```
PerformWithRetry hit unexpected exception System.Net.Http.HttpRequestException
Stack:
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Fabric.FabricDeployer.ContainerNetworkDeploymentOperation.<>c__DisplayClass17_0.<IsWireServerAvailable>b__0()
   at System.Fabric.Common.Helpers.PerformWithRetry(Action action, Type[] exceptionTypes, TimeSpan retryInterval, Int32 retryCount)
```

**Attached Log files:**
[Log.zip](https://github.com/Azure/service-fabric-issues/files/2481125/Log.zip)

**Comment by dharmeshtailor** [comment]
So here is another update. My local Service fabric is working perfectly now (for the time being. Fingers crossed).

When I ran the clean and setup dev cluster scripts, I got usual error while starting the service.

But then I started local cluster manager and started the cluster and execute my code and it worked all perfect.

There are couple of things I did though:

Under **Hosting**:
```
{
    "name": "FabricContainerAppsEnabled",
    "value": "false"
}
```

And under **Nodes**, I replaced "**iPAddress**" to "**127.0.0.1**"

```
"nodes": [
    {
      "nodeName": "_Node_0",
      "iPAddress": "127.0.0.1",
      "nodeTypeRef": "NodeType0",
      "faultDomain": "fd:/0",
      "upgradeDomain": "0"
    }
  ]
```

I am not sure, which on this fixed the issue. But for now I am happy. Let me monitor this for some more time and I'll report back here.

**Comment by dragav** [comment]
Thank you for the update, as well as the hint. Will keep the issue open until we roll out a fix.

**Comment by dharmeshtailor** [comment]
Just an update, "Reset Local Cluster" still fails with error, but when I execute my code, it gets deployed and work properly.

**Comment by marclallen** [comment]
Sorry for the delay.  Other fires. :)

I see some progress has been made.  I'll try these fixes and if they don't work, I'll post a traces log.

**Comment by SudhindraKovalam** [comment]
One thing that helped me fix my issue, is checking the `clusterManifest.xml` file in `C:\SfDevCluster\Data` folder. check the property 
```
 <Infrastructure>
    <WindowsServer IsScaleMin="true">
      <NodeList>
        <Node NodeName="_Node_0" IPAddressOrFQDN="XYZ" IsSeedNode="true" NodeTypeRef="NodeType0" FaultDomain="fd:/0" UpgradeDomain="0" />
      </NodeList>
    </WindowsServer>
  </Infrastructure>
```
if the value in `IPAddressOrFQDN` doesnot match the value shown in your computer full name, this could also result in the error above.

**Comment by shsha-msft** [comment]
Thanks a lot to everyone who reported this issue and provided their inputs. We have fixed it in our development branch and the fix will be released in a future update of Service Fabric Runtime.

Since discussions on this thread involved a few different errors I want to clarify that we have fixed the E_ACCESSDENIED issue which can be worked around by either.

1. Setting FabricContainerAppsEnabled to false or,
2. Providing NETWORK SERVICE write access to C:\ProgramData\Microsoft\Crypto\RSA\MachineKeys

I am closing this particular issue. If you are experiencing errors which cannot be worked around by the the two options mentioned above then it is likely to be a different problem and so please feel free to create a new GitHub issue.

**Comment by dhaval24** [comment]
> We got this problem on almost all dev computers when upgrading to SF 3.2.176.
> Adding write permission for NETWORK SERVICE or even removing the security config for NETWORK SERVICE resolves the issue ("Everyone" has Write permission).
> 
> One dev computer was set up from scratch with SF 3.2.176 as first the installed SDK. Not sure, but on this computer I think this issue emerged when using the Refresh debug application mode (which creates a couple security groups). I'm pretty sure it worked starting a cluster before that, so maybe a combination the activating Refresh and the update.
> 
> By the way, running a secure cluster.

@rnilsark  I was facing similar issue as well when connecting with secure SF cluster with SF version 4.0.470. This helped me fix the issue too. Thanks
